docs/.vuepress/tools/check-examples/helpers.js

```diff
@@ -0,0 +1,154 @@
+ * @param {boolean} warn `true` if the check was successful, `false` otherwise.
+ */
+function logCheck(valid, warn) {
+  // eslint-disable-next-line no-nested-ternary
+  process.stdout.write(chalk[(valid ? (warn ? 'yellow' : 'green') : 'red')]('.'));
+}
+
+/**
+ * Wait for the amount of milliseconds passed as an argument.
+ *
+ * @param {number} timeout Number of milliseconds to wait.
+ * @returns {Promise}
+ */
+async function sleep(timeout) {
+  return Promise.resolve({
+    then(resolve) {
+      setTimeout(resolve, timeout || 300);
+    }
+  });
+}
+
+/**
+ * Make the provided string to have the first letter uppercased.
+ *
+ * @param {string} string String to process.
+ * @returns {string}
+ */
+function firstUppercase(string) {
+  return string.charAt(0).toUpperCase() + string.slice(1);
+}
+
+/**
+ * Serve the pre-build documentation files.
+ *
+ * @param {number} PORT The port for the files to be served on.
+ */
+function serveFiles(PORT) {
+  spawnProcess(`http-server ${path.resolve(__dirname, '../../dist')} -s -p ${PORT}`, {});
+}
+
+/**
+ * Go through the `.md` files looking for `::: example` containers and return the search results.
+ *
+ * @param {string} version Version of the docs to look for in.
+ * @returns {Promise}
+ */
+async function findExampleContainers(version) {
+  return replaceInFiles({
+    files: path.join(mdDir, version, '**/*.md'),
+    from: /::: example/g,
+    onlyFindPathsWithoutReplace: true
+  });
+}
+
+/**
+ * Setup the headless browser.
+ *
+ * @returns {Promise<{browser: Browser, page: Page}>}
+ */
+async function setupBrowser() {
+  const browser = await puppeteer.launch({
+    headless: false,
+    args: ['--headless', '--disable-gpu', '--mute-audio'],
+  });
+  const page = await browser.newPage();
+
+  return {
+    browser,
+    page
+  };
+}
+
+/**
+ * Get the permalinks of all the pages containing the `::: example` container.
+ *
+ * @param {object} searchResults Search results returned from the `findExampleContainers` function.
+ * @returns {string[]}
+ */
+function fetchPermalinks(searchResults) {
+  const permalinks = [];
+
+  searchResults.paths.forEach((filePath) => {
+    const mdFile = fs.readFileSync(filePath, {
+      encoding: 'utf8'
+    });
+
+    const permalink = /permalink: (.*)\n/g.exec(mdFile)[1];
+
+    permalinks.push(permalink);
+  });
+
+  return permalinks;
+}
+
+/**
+ * Extend the permalink with the framework information and modify the url to match the versioned structure (the
+ * latest version is available as an url without a version number).
+ *
+ * @param {string} permalink The permalink to be modified.
+ * @param {string} framework The framework to be processed.
+ * @param {string} version The version to be processed.
+ * @returns {string}
+ */
+function extendPermalink(permalink, framework, version) {
+  const latestVersion = getLatestVersion();
+  const frameworkSpecificDocsVersions = getDocsFrameworkedVersions('development');
```

Comment:
Should it be hardcoded?

Reply:
It's used just to retrieve the list of all "framework-specific" documentation versions (including `next`), so I think it's fine here.